MailManager/src/main/java/inc/softserve/health/MailManagerHealthCheck.java

```diff
@@ -30,33 +33,42 @@ public MailManagerHealthCheck(@ESPConnectorAdmin String ESPConnectorAdminUrl,
         this.mongo = mongo;
     }
 
-    protected Result check() throws Exception {
-        boolean espConnectorHealthy = false;
-        StringBuilder unhealthyMessage = new StringBuilder("");
+    protected Result check() {
+        ExecutorService executor = Executors.newFixedThreadPool(3);
+        FutureTask<Boolean> espConnectorCheck = new FutureTask(this::isEspConnectorHealthy);
+        FutureTask<Boolean> mongoCheck = new FutureTask(this::isMongoHealthy);
 
+        FutureTask<Boolean> commonHealth = new FutureTask(() -> {
+            executor.execute(espConnectorCheck);
+            executor.execute(mongoCheck);
+            return espConnectorCheck.get() && mongoCheck.get();
+        });
+
+        try {
+            executor.execute(commonHealth);
+            return (commonHealth.get(500L, TimeUnit.MILLISECONDS)) ? Result.healthy() : Result.unhealthy("Service unavailable!");
```

Comment:
That is preferable to split in three lines: condition, true and false statements.